File: solutions/_1700_number_of_students_unable_to_eat_lunch/solution.js
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function (students, sandwiches) {
  let studentsRemaining = students.length;
  const count = [0, 0];
  for (const pref of students) {
    count[pref]++;
  }

  for (const sandwich of sandwiches) {
    if (count[sandwich] === 0) break;
    count[sandwich]--;
    studentsRemaining--;
  }

  return studentsRemaining;
};

module.exports = { countStudents };

/*
https://leetcode.com/problems/number-of-students-unable-to-eat-lunch/
*/

File: solutions/_1700_number_of_students_unable_to_eat_lunch/solution.py
from typing import List


class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        students_left = len(students)
        count = [0, 0]
        for pref in students:
            count[pref] += 1

        for sandwich in sandwiches:
            if count[sandwich] == 0:
                break
            students_left -= 1
            count[sandwich] -= 1

        return students_left


# https://leetcode.com/problems/number-of-students-unable-to-eat-lunch/
